lib/TimeoutCancellationToken.php

```diff
@@ -15,14 +15,15 @@ final class TimeoutCancellationToken implements CancellationToken
 
     /**
      * @param int $timeout Milliseconds until cancellation is requested.
+     * @param string $message Message for TimeoutException. Default is "Operation timed out".
      */
-    public function __construct(int $timeout)
+    public function __construct(int $timeout, string $message = "Operation timed out")
     {
         $source = new CancellationTokenSource;
         $this->token = $source->getToken();
 
-        $this->watcher = Loop::delay($timeout, static function () use ($source) {
-            $source->cancel(new TimeoutException);
+        $this->watcher = Loop::delay($timeout, static function () use ($source, $message) {
+            $source->cancel(new TimeoutException($message));
```

Comment:
Should it be possible to provide the whole exception instead?

Reply:
I don't think I care for that, since then the instance has to be made prior to the timeout (unless we used a callable, but then again, more complexity for something simple).